From 457a95cf917cfc030bafd2da5d943e1b93051afe Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Fri, 11 Jul 2014 00:34:58 +0100
Subject: [PATCH 15/22] session: Fix memory issues

Ensure we free strings coming back from dbus and don't unref ourselves
when GTK will already do that.
---
 session/budgie-session-dialog.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/session/budgie-session-dialog.c b/session/budgie-session-dialog.c
index 68719c4..b70f768 100644
--- a/session/budgie-session-dialog.c
+++ b/session/budgie-session-dialog.c
@@ -90,6 +90,8 @@ static void budgie_session_dialog_init(BudgieSessionDialog *self)
                         if (response == SD_YES || response == SD_CHALLENGE) {
                                 can_reboot = TRUE;
                         }
+                        g_free(result);
+                        result = NULL;
                 }
                 /* Can we shutdown? */
                 if (!sd_login_manager_call_can_power_off_sync(self->proxy,
@@ -100,10 +102,11 @@ static void budgie_session_dialog_init(BudgieSessionDialog *self)
                         if (response == SD_YES || response == SD_CHALLENGE) {
                                 can_poweroff = TRUE;
                         }
+                        g_free(result);
+                        result = NULL;
                 }
         }
 
-
         gtk_window_set_position(GTK_WINDOW(self), GTK_WIN_POS_CENTER_ALWAYS);
         gtk_window_set_skip_taskbar_hint(GTK_WINDOW(self), TRUE);
         gtk_window_set_skip_pager_hint(GTK_WINDOW(self), TRUE);
@@ -222,7 +225,6 @@ static void clicked(GtkWidget *button, gpointer userdata)
                         g_message("Unable to logout!");
                 }
         }
-        g_object_unref(self);
         gtk_main_quit();
 }
 
-- 
1.9.1

